front/app/components/InitiativeMap/components/StartInitiativeButton.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { createPortal } from 'react-dom';
+
+// components
+import { Button } from '@citizenlab/cl2-component-library';
+
+// intl
+import { useIntl } from 'utils/cl-intl';
+import messages from '../messages';
+
+type Props = {
+  modalPortalElement: HTMLDivElement;
+  onClick: () => void;
+};
+
+const StartInitiativeButton = ({ modalPortalElement, onClick }: Props) => {
+  const { formatMessage } = useIntl();
+  if (!modalPortalElement) return null;
+
+  return createPortal(
```

Comment:
Same suggestion here- I think we only should use `createPortal` for full screen things

Reply:
@luucvanderzee This one is actually still needed I think :thinking: When I was looking in the Esri forums, this was the recommended way to customize the Esri map popup with custom React components.

We create a div element in the code and then assign it to the Esri popup's `popup.content`, and then we can use a portal to render whatever React components within the popup content.

E.g. This button can only be rendered within the popup by using the portal:

![image](https://github.com/CitizenLabDotCo/citizenlab/assets/33987955/5841a5a5-83b2-4958-b567-60cb44946f07)